src/plugins/link/src/main/js/ui/Dialog.js

```diff
@@ -383,10 +384,25 @@ define(
         };
       }
 
-      if (Settings.hasLinkClassList(editor.settings)) {
+      if (editor.settings.style_formats) {
+        var classList;
+
+        classList = CascadeUtils.getClassesForDropdown(editor, editor.settings.link_class_list);
+
+        if (classList) {
+          // Add a 'None' option to the beginning if it is not already present.
+          if (typeof classList[0] !== 'object') {
```

Comment:
I kept trying to extract this out by using `CascadeUtils.appendOptionsToDropDown(classList, data);` (same as in the image plugin), but `grunt` kept throwing syntax errors for some reason when I tried to compile the change. I gave up temporarily to keep the issue moving 😞

Reply:
We worked through this with a "common plugin"